### DISCUSSION TITLE
Improve IDE support, which should be seen as part of Grails project itself

### DISCUSSION THREAD

**Discussion by kico-itexto** [apache/grails-core#13546] [Ideas]
This is a MAJOR problem on the Groovy ecossystem and should be thought as part of the Grails project if we want to gain traction.

Grails was initially projected to allow developers to work using only the CLI and a simple code editor: the problem is that we live on a JVM world, where we must deal with all the APIs provided by the libraries and plugins that compose our projects.

Until Grails 2.x we could rely on the support of some free IDEs like the old GGTS and Netbeans, but this is not the reality. There were some talks saying that "all major IDEs have some Grails support because of Gradle" (when on Grails 3 the old Gant was retired), but today that's not the case (actually Gradle support has been always many steps behind what we can find for Maven). We can see some support on the PAID version of Intellij and basically this is all we get today.

If we try to use something like VS Code (which is maybe more mainstream today than Eclipse) we see that we have almost no support both for Groovy and even less for Grails (we only have some code highlight).

Here is the situation of the code-groovy plugin for VS Code:

![image](https://github.com/user-attachments/assets/957300c8-d766-4b5f-86ba-c7f861370b28)

the last release is dated way back to 2018! And this is the most popular one.

**Comment by gsartori** [comment]
I agree on this.

At least for marketing/communication purposes we should update all the Groovy & Grails IDE plugin's "last update" metadata with the current year. Even if it means to just add a blank line to a random class and re-deliver.

The first thing junior developers watch is the "last updated" date.

That alone is a go/no-go decision.

**Comment by jamesfredley** [comment]
There was solid open source support for Grails in Eclipse and NetBeans, until several years ago.   It would be nice to see that picked back up for Eclipse and VS Code, which have decent market share.  

For now, IntelliJ IDEA has been the #1 Java IDE for many years, passing Eclipse and NetBeans, which was no small feat, and has maintained that lead.

The IntelliJ Grails plugin is constantly updated.  https://plugins.jetbrains.com/plugin/18504-grails/versions

IntelliJ has historically provided the highest level of support for Groovy and Grails and it continues to do so today.   The Grails plugin is only available with the paid ultimate version and in my experience it is worth the small cost for the large developer productivity increases, across several teams I have worked on.   There are many ways to get a lower price through JetBrains and the price goes down with each annual renewal. 

In 2024, IntelliJ is the way to go for Grails development and is what most engineers are using.  I used NetBeans and Eclipse for about 15 years combined before fully switching to IntelliJ, with years of overlap between the three.

**Reply by jamesfredley** [reply]
Surveys with Java IDE market share.   

https://www.baeldung.com/java-ides-in-2024

https://snyk.io/blog/intellij-idea-dominates-the-ide-market-with-62-adoption-among-jvm-developers/

![Java IDEs 2024](https://github.com/user-attachments/assets/91133b5a-c77d-4566-829e-6b0dd0786609)

![Java IDEs 2019](https://github.com/user-attachments/assets/62402018-9a68-4227-9b76-b39514a10370)


![https___res cloudinary com_snyk_image_upload_v1589486428_25b](https://github.com/user-attachments/assets/d94571ad-daa9-47a7-b6f8-c7ce023fbe89)

**Reply by codeconsole** [reply]
> The IntelliJ Grails plugin is constantly updated. https://plugins.jetbrains.com/plugin/18504-grails/versions

@jamesfredley Is it constantly updated? or is it just rebuilt with the latest version of the IDE?  It's hard to believe any updates have happened in quite a long time considering it has been broken for a while now with the latest versions of Grails.

**Reply by jamesfredley** [reply]
I am not sure how many of those are rebuilds against a new IDE version.  

I've used the plugin consistently for the last 3 years and have only bumped into 1 or 2 show stopper issues after new Grails releases that caused me to wait on an IDE or Grails release, usually for a few weeks.  I am normally on the latest version of IntelliJ and ran EAP for a bit during one release.  During those 3 years I have seen a number of plugin issues resolved and features added.  Sometimes there has been a hunt to find a solution, like the grails-shell dependency with Grails 6+.

Some issues haven't impacted me as much as other teams.

**Reply by logantracyo** [reply]
@jamesfredley and all, it appears that JetBrains is planning to drop support for their plugin:

Issue created on 24-Sep-2025 -- https://youtrack.jetbrains.com/issue/IDEA-379825/Move-Grails-to-obsolete-plugins
"We propose to mark the Grails IntelliJ plugin as obsolete. The plugin is no longer actively used, and we would like to stop building and distributing it as part of our regular release process."

It has not yet been moved, but their plan is:
"Anyone interested can fork the existing codebase and continue development independently. See https://github.com/JetBrains/intellij-obsolete-plugins"

You'll need a YouTrack account to comment, whether asking JetBrains to continue development, or to propose supporting community development, etc.

**Reply by jamesfredley** [reply]
I have replied on https://youtrack.jetbrains.com/issue/IDEA-379825/Move-Grails-to-obsolete-plugins and hope to hear back, one way or the other.

**Reply by jamesfredley** [reply]
After a quick conversation with JetBrains this morning, there should be no concern about the future of this plugin.  The plugin id will stay the same going forward and we will iron out the details around where the code lives and who manages PRs and releases over the coming months.

In IntelliJ 2026.1, which is due in March, the Grails plugin will continue to work.

**Comment by rainboyan** [comment]
I don't know if you have ever thought that since Grails is built on Spring Boot, then a Grails application must also be a Spring Boot application, but why can't we just use Spring Tools?

I think IntelliJ IDEA Ultimate, [STS 4 for Eclipse](https://spring.io/tools) and [Groovy Eclipse](https://github.com/groovy/groovy-eclipse) would be a good way to develop and debug Grails applications, of course, we prefer VS Code, and if I have more time in the future, I'll try to develop a Groovy & Grace extension for it.

<img width="912" alt="grace_as_spring_boot_application_02" src="https://github.com/user-attachments/assets/c5cef0b0-aed6-41ee-8702-5ba581390b5b">

<img width="1512" alt="grace_as_spring_boot_application_04" src="https://github.com/user-attachments/assets/03ee5f25-d7cf-4482-a71e-9dd3c571f0e6">

**Comment by jdaugherty** [comment]
FYI: We created https://github.com/grails/grails-core/issues/13910 to track various "known" IntelliJ IDE issues.

**Reply by jamesfredley** [reply]
IntelliJ 2025.1 has been released with several Grails and Groovy related updates

https://blog.jetbrains.com/idea/2025/04/intellij-idea-2025-1/


https://youtrack.jetbrains.com/articles/IDEA-A-2100662421/IntelliJ-IDEA-2025.1-251.23774.435-build-Release-Notes


https://youtrack.jetbrains.com/issue/IDEA-342753/IDEA-does-not-recognize-dependencies-between-Modules-in-a-Grails-Project
https://youtrack.jetbrains.com/issue/IDEA-357902/Deprecated-property-configuration-is-not-highlighted-in-Groovy-build-files
https://youtrack.jetbrains.com/issue/IDEA-360448/Exception-in-Groovy-plugin-on-invocation-of-Search-everywhere
[https://youtrack.jetbrains.com/issue/IDEA-363716/Groovy-Parameter-info-shows-Object...-a[](https://youtrack.jetbrains.com/issue/IDEA-363716/Groovy-Parameter-info-shows-Object...-a%5B)…]nd-no-parameters-instead-of-actual-signatures-in-closures
https://youtrack.jetbrains.com/issue/IDEA-358458/Adding-JSpecify-Nullable-to-Groovy-method-incorrectly-triggers-an-error
BugKTIJ-32962Gradle multi-module wizard collects wrong data when using Groovy

**Comment by kico-itexto** [comment]
Hope they hear you.

But today, with tools like Github Copilot (which will supply what we need
for auto complete at least), is this plugin still relevant?

Em seg., 23 de fev. de 2026 às 23:50, James Fredley <
***@***.***> escreveu:

> I have replied on
> https://youtrack.jetbrains.com/issue/IDEA-379825/Move-Grails-to-obsolete-plugins
> and hope to hear back, one way or the other.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/grails-core/discussions/13546#discussioncomment-15903949>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6W7AARAYKBJH65NMAXZMD4NO4BBAVCNFSM6AAAAACV5ITI5SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKOJQGM4TIOI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Henrique Lobo Weissmann
Diretor Técnico
<https://www.linkedin.com/company/itexto>
*Tel:* +55 (31) 2573-1979 <+55(31)2573-1979>
*Email:* ***@***.***
*Web:* www.itexto.com.br

**Comment by kico-itexto** [comment]
Does it mean there is a possibility of making this plugin available on the
Community Edition of IDEA?

Em sex., 27 de fev. de 2026 às 19:10, James Fredley <
***@***.***> escreveu:

> After a quick conversation with JetBrains this morning, there should be no
> concern about the future of this plugin. The plugin id will stay the same
> going forward and we will iron out the details around where the code lives
> and who manages PRs and releases over the coming months.
>
> In IntelliJ 2026.1, which is due in March, the Grails plugin will continue
> to work.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/grails-core/discussions/13546#discussioncomment-15951422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6W7ABOEXBSOY5KQIQIX5D4OC6FPAVCNFSM6AAAAACV5ITI5SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKOJVGE2DEMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Henrique Lobo Weissmann
Diretor Técnico
<https://www.linkedin.com/company/itexto>
*Tel:* +55 (31) 2573-1979 <+55(31)2573-1979>
*Email:* ***@***.***
*Web:* www.itexto.com.br

**Reply by jamesfredley** [reply]
We should have more info in the next 1-2 months.
